### ISSUE TITLE
[Bug]: aws_cloudwatch_log_group resource: Allow 3 year retention

### ISSUE THREAD

**Issue by leewilmott** [hashicorp/terraform-provider-aws#29423]
### Terraform Core Version

1.3.8

### AWS Provider Version

4.51.0

### Affected Resource(s)

aws_cloudwatch_log_group

### Expected Behavior

I successfully changed the retention settings in the console to 3 years (1096 days).

I went to do this via Terraform and I am unable to do it.  The documentation says that only the following is allowed for the retention_in_days attribute:

1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 2192, 2557, 2922, 3288, 3653, and 0

I would have expected Terraform to allow the same values as per the console.

### Actual Behavior

I am shown an error:

```
Error: expected retention_in_days to be one of [0 1 3 5 7 14 30 60 90 120 150 180 365 400 545 731 1827 2192 2557 2922 3288 3653], got 1096
│
│   with module.customer_contoso.aws_cloudwatch_log_group.flow_log_group,
│   on customer/flow_logs.tf line 11, in resource "aws_cloudwatch_log_group" "flow_log_group":
│   11:   retention_in_days = var.flow_log_retention_in_days
```

### Terraform Configuration Files

```hcl
resource "aws_flow_log" "flow_log" {
  iam_role_arn    = var.flow_log_role_arn
  log_destination = aws_cloudwatch_log_group.flow_log_group.arn
  traffic_type    = "ALL"
  vpc_id          = aws_vpc.main.id
}

resource "aws_cloudwatch_log_group" "flow_log_group" {
  name = "testing-flowlogs"
  retention_in_days = 1096
}
```

### Steps to Reproduce

Created the following resource:

```hcl
resource "aws_cloudwatch_log_group" "flow_log_group" {
  name = "testing-flowlogs"
  retention_in_days = 1096
}
```

...and applied it in Terraform.

### Would you like to implement a fix?

Yes

**Comment by leewilmott** [comment]
Hi @mattburgess, many thanks for commenting so soon!

Yes, I would love to have a go at implementing a fix...it's my first but I'd like to give it a go if that's ok.  I've just been reading up on the documentation and I will try shortly.

**Comment by leewilmott** [comment]
Hi @mattburgess,

All the links you sent me were very interesting; as you said, no where does it mention that 1096 is a valid value.

I created the following PR to hopefully fix this issue...

https://github.com/hashicorp/terraform-provider-aws/pull/29426

Also, I specifically created a test to check the 1096 value...I don't know whether this is necessary, but I thought I would add it because if 1096 is not documented then 'perhaps' there is a chance that it could be removed.  Interested to know your thoughts on this.

I'm wondering if I created the .changelog file with the correct name...the filename I used was the name of the number of this ticket...is that right?

Many thanks,

Lee
